### PR TITLE
update the rdkit-postgresql recipe for release 2018.03

### PR DESCRIPTION
**Pull request by rvianello**
this is still in progress, but the current changes should allow building the cartridge for linux using the postgresql 10.3 package from the anaconda distribution, so it might be also sufficient for performing some preliminary tests on osx too.

next steps should include proper versioning of the postgresql dependency and building for both postgres 10 and 9, plus some more in depth testing.

eventually, I would like to remove the bundled postgresql recipes, and the rdkit-postgresql95 recipe (no longer needed and made unnecessary by the improved support for build variants in conda-build 3+).

### CODE REVIEW

**Comment by greglandrum** [comment]
Thanks Riccardo.
I will try and take a look on the Mac either this evening or tomorrow morning.

File: rdkit-postgresql/cmakelists.txt.patch [hunk 1]
```diff
@@ -1,18 +1,68 @@
...
 +
++set(CMAKE_MODULE_PATH
++    ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/Modules/")
++
 +find_package(RDKit REQUIRED)
 +include_directories(${RDKit_INCLUDE_DIRS})
 +
-+find_package(PostgreSQL REQUIRED)
++# Boost and Eigen libraries that are required at link time
++find_package(Boost 1.56.0 COMPONENTS serialization thread)
```
Comment (greglandrum) [inline rdkit-postgresql/cmakelists.txt.patch]: You should be able to drop thread from the requirements here.

**Comment by greglandrum** [comment]
If you were willing to do an actual RDKit build here, you could almost certainly get rid of the runtime RDKit and boost dependencies by linking against each statically. Seems like this could be quite useful.

**Comment by greglandrum** [comment]
I did a Mac build this morning and everything worked fine, but I'm kind of curious about the output. I got the following:
```
anaconda upload /Users/glandrum/anaconda5/conda-bld/osx-64/rdkit-postgresql-2018.03.1.1-h4feeb9f_1.tar.bz2
anaconda upload /Users/glandrum/anaconda5/conda-bld/osx-64/rdkit-postgresql-2018.03.1.1-hec0650b_1.tar.bz2
anaconda upload /Users/glandrum/anaconda5/conda-bld/osx-64/rdkit-postgresql-2018.03.1.1-hb8657ce_1.tar.bz2
```
I assume that each of the builds is against a different version of postgresql, but I'm surprised that the psql version doesn't somehow show up in the filename. If that could be added, I think it would make what has been built a bit clearer.

File: rdkit-postgresql/cmakelists.txt.patch (hunk 1 above)
Comment (rvianello) [reply rdkit-postgresql/cmakelists.txt.patch]: I just tried, but cmake produced a warning reading `[...] Target "rdkit" links to target "Threads::Threads" but the target was not found.  Perhaps a find_package() call is missing for an IMPORTED target, or an ALIAS target is missing?` and the build failed with `/home/ric/rdkb/rdkit-postgresql_1525713265598/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.2.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lThreads::Threads`

**Comment by rvianello** [comment]
running a full rdkit build, and this way statically link some dependencies, would also get rid of ensuring that the dependency from rdkit is correctly tracked (which is currently quite difficult to do). I will give it a try.

regarding the dependency of different build variants from distinct versions of postgres, it's actually already reflected in the filename, in fact it contributes to the hash string that conda-build 3 uses to summarize all the recipe metadata (e.g. h4feeb9f). this is definitely not human readable, and not as friendly as a different package name (e.g. rdkit-postgresql95) would be, but it's so by design, because a different package name would be interpreted as a completely unrelated component. with this (somewhat questionable) generalization of build strings, it is easy to create a package X that depends on rdkit-postgresql, without referencing any particular version of postgres, this way allowing the same package X to be used with any version of postgresql, as long as a suitable build variant of rdkit-postgresql is found (with the older approach, a different X-Y package would have been required to be built for every rdkit-postgresqlY).

**Comment by greglandrum** [comment]
@rvianello : do you want me to wait a bit longer before merging this?

**Comment by rvianello** [comment]
@greglandrum merging would be fine for me, if you are ok with leaving the windows support on hold for now (in the current revision it's no longer functional), and I would be happy to use smaller, separate PRs to propose a cleanup of the recipes that these changes are obsoleting (e.g. rdkit-postgresql95).

**Comment by greglandrum** [comment]
Sounds good. I agree that doing the other cleanups in smaller PRs makes sense.

**Pull request merged**